src/test/java/com/google/api/generator/engine/ast/NewObjectValueTest.java

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.api.generator.engine.ast;
+
+import static junit.framework.Assert.assertEquals;
+
+import java.awt.List;
+import java.util.Arrays;
+import java.util.Collections;
+import org.junit.Test;
+
+public class NewObjectValueTest {
+  @Test
+  public void simpleNewObjectValue() {
+    ConcreteReference reference =
+        ConcreteReference.builder()
+            .setClazz(List.class)
+            .setGenerics(Collections.emptyList())
+            .build();
+    TypeNode type = TypeNode.withReference(reference);
+    NewObjectValue newObjectValue = NewObjectValue.genericBuilder().setType(type).build();
+    assertEquals(newObjectValue.type(), type);
+    assertEquals(newObjectValue.value(), "new List<>");
+  }
+
+  @Test
+  public void newObjectValue_generics() {
+    ConcreteReference reference =
+        ConcreteReference.builder()
+            .setClazz(List.class)
+            .setGenerics(Arrays.asList(ConcreteReference.withClazz(String.class)))
+            .build();
+    TypeNode type = TypeNode.withReference(reference);
+    NewObjectValue newObjectValue = NewObjectValue.genericBuilder().setType(type).build();
+    assertEquals(newObjectValue.type(), type);
+    assertEquals(newObjectValue.value(), "new List<String>");
+  }
+
+  @Test
+  public void newObjectValue_withArgs() {
+    ConcreteReference reference = ConcreteReference.withClazz(Integer.class);
+    TypeNode type = TypeNode.withReference(reference);
+    ValueExpr valueExpr =
+        ValueExpr.builder()
+            .setValue(PrimitiveValue.builder().setType(TypeNode.INT).setValue("123").build())
+            .build();
+    NewObjectValue newObjectValue =
+        NewObjectValue.builder().setType(type).setArguments(Arrays.asList(valueExpr)).build();
+    assertEquals(newObjectValue.type(), type);
+    assertEquals(newObjectValue.value(), "new Integer");
+  }
+}
```

Comment:
Could we please add a test with more arguments?

Reply:
Also, a test that exercises the failure case.

Reply:
Also, tests that *fully* exercise the `isGeneric() && generics().isEmpty()` logic.

Reply:
sure, the test with args is in JavaWriterVisitorTest, and a failure test is added to the NewObjectValue test, because of the conflict setting of `isGeneric()` and `type().reference.generics()`.

Reply:
On reading the usage of the API in this test, can you see anything that might look confusing to a user? How many edge cases does that method's logic have, and what are the side effects?

As a user, I would expect that this usage should not error out. Could you please write more unit tests that *fully* exercise the edge case logic? This can help us find unexpected behavior and change the implementation.

Reply:
Alternatively, you may leave a TODO here for me and I can fix it if that helps.